Added new Fine-Tuning a smol VLM (SmolVLM) with the Hugging Face Ecosystem (TRL) using SFT on a Consumer GPU recipe 🧑‍🍳️

sergiopaniego opened the pull request:
# What does this PR do?

<!--
Thank you for submitting a PR to the Open-source AI cookbook!

Someone will review your PR shortly. They may suggest changes to further improve your contribution. 
If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same 
persons---sometimes notifications get lost.

-->

<!-- Remove if not applicable -->

Fixes #245

## Who can review?

@merveenoyan and @stevhliu.

cc. @andimarafioti @burtenshaw

<!-- 
At the moment, please tag @merveenoyan and @stevhliu.
-->

stevhliu reviewed (approved):
Great work! :)

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/cookbook/pr_246). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

sergiopaniego commented:
<p>Updated!</p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/246/discussion/#comment-2523868117'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

sergiopaniego commented:
<p>When using <code>train_dataset.map(format_data)</code>, the <code>format_data</code> function is expected to return a dictionary, not a callable function.</p><p> </p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/246/discussion/#comment-2521347415'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

sergiopaniego commented:
<p>In the current PR file, only one is displayed: [View here](<a href="https://github.com/huggingface/cookbook/blob/51c9a13a8409e773febb79f3e3ee127bcbe672bc/notebooks/en/fine_tuning_smol_vlm_sft_trl.ipynb" target="_blank">https://github.com/huggingface/cookbook/blob/51c9a13a8409e773febb79f3e3ee127bcbe672bc/notebooks/en/fine_tuning_smol_vlm_sft_trl.ipynb).</a></p><p> </p><p>I'm not sure if this might cause an issue when rendered after upload.</p><p> </p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/246/discussion/#comment-2523868140'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

sergiopaniego commented:
I've updated the notebook accordingly, @burtenshaw @stevhliu! 😄  

I've also added a note at the beginning regarding the dataset used. The dataset was utilized during the training of the instruct model, meaning the model is already familiar with it. However, it can still be used for educational purposes.

stevhliu commented on notebooks/en/_toctree.yml:
```diff
@@ -94,6 +94,8 @@
         - local: multimodal_rag_using_document_retrieval_and_reranker_and_vlms
           title: Multimodal RAG with ColQwen2, Reranker, and Quantized VLMs on Consumer GPUs
+        - local: fine_tuning_smol_vlm_sft_trl
+          title: Fine-Tuning a smol VLM (SmolVLM) with the Hugging Face Ecosystem (TRL) using SFT on a Consumer GPU
```
Let's also update the title here to ensure they match :)

stevhliu commented on notebooks/en/index.md:
```diff
@@ -7,11 +7,11 @@ applications and solving various machine learning tasks using open-source tools
 - [Fine-Tuning a Vision Language Model (Qwen2-VL-7B) with the Hugging Face Ecosystem (TRL)](fine_tuning_vlm_trl)
 - [Multi-agent RAG System 🤖🤝🤖](multiagent_rag_system)
 - [Multimodal RAG with ColQwen2, Reranker, and Quantized VLMs on Consumer GPUs](multimodal_rag_using_document_retrieval_and_reranker_and_vlms)
+- [Fine-Tuning a smol VLM (SmolVLM) with the Hugging Face Ecosystem (TRL) using SFT on a Consumer GPU](fine_tuning_smol_vlm_sft_trl)
```
And update here as well

sergiopaniego replied:
Ouch, my bad. Updated!

merveenoyan reviewed (commented):
Left some nits, thank you Sergio!

sergiopaniego commented:
Thanks a lot for the comments @merveenoyan! I've updated the notebook accordingly. Since the title has been modified, I've also updated `notebooks/en/_toctree.yml` and `notebooks/en/index.md` files-
